### PR TITLE
Generate sitemap.xml + robots.txt

### PR DESCRIPTION
**Pull request by MattIPv4**
Please let me know if you disagree with any of the routes I'm excluding from either the sitemap or robots.

### CODE REVIEW

**Comment by MattIPv4** [comment]
I've set `NEXT_PUBLIC_NOINDEX` to `true` for production + previews for now -- once we do the switch over we'll want to unset it for production.

File: apps/website/next-sitemap.config.cjs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  // Create a robots txt file, and disallow indexing of some routes
+  generateRobotsTxt: true,
+  robotsTxtOptions: {
+    policies: [
+      {
+        userAgent: "*",
+        disallow:
+          process.env.NEXT_PUBLIC_NOINDEX === "true"
+            ? ["/"]
+            : ["/api/", "/auth/", "/admin/", "/wip/", "/giveaways/"],
```
Comment (pjeweb) [inline apps/website/next-sitemap.config.cjs:32]: can we add `/privacy-policy` here as well? there are some people searching for "bad" texts policies on search engines trying to sue people (at least here in germany).

File: apps/website/next-sitemap.config.cjs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "/api/*",
+    "/auth",
+    "/auth/*",
+    "/admin",
+    "/admin/*",
+    "/wip",
+    "/wip/*",
+    "/giveaways",
+    "/giveaways/*",
+    "/show-and-tell/*",
```
[the remark this answers is not in the document]
Comment (MattIPv4) [reply apps/website/next-sitemap.config.cjs:20]: Yah, and I honestly think it probably does more harm than good to have them indexed as they'll take away value from the main site pages

File: apps/website/next-sitemap.config.cjs (hunk 1 above)
Comment (MattIPv4) [reply apps/website/next-sitemap.config.cjs:32]: Sure

**Comment by pjeweb** [comment]
can we merge this?

**Comment by pjeweb** [comment]
sorry broke it with the other PR :(

**Review by pjeweb — APPROVED** [review]

**Pull request merged**